torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -119,68 +100,29 @@ void IRReplaceSizes() {
     }
   }
 
-  // If we already lowered all inputs/outputs we can just return.
-  if (size_map.size() == 0)
-    return;
-
-  // Set domains to be based on symbolic sizes (i.e. Ti.size[...])
-  for (TensorView* tv : all_tvs) {
-    std::vector<IterDomain*> new_domain_iters;
-    const std::vector<IterDomain*>& root_td = tv->getRootDomain();
-
-    for (decltype(root_td.size()) i{0}; i < root_td.size(); i++) {
-      Val* new_size = root_td[i]->extent();
-      if (size_map.find(new_size) != size_map.end())
-        new_size = size_map[new_size];
-
-      new_domain_iters.push_back(new IterDomain(
-          root_td[i]->start(),
-          new_size,
-          root_td[i]->parallel_method(),
-          root_td[i]->isReduction(),
-          root_td[i]->isRFactorProduct(),
-          root_td[i]->isBroadcast()));
-    }
-
-    TensorDomain* old_domain = tv->domain();
-    TensorDomain* new_domain = new TensorDomain(new_domain_iters);
-
-    // We should just be able to replace sizes in place, but mutator is setup to
-    // do that as it set up to replace vals in Exprs, but
-    // IterDomain/TensorDomain are vals.
-
-    new_domain = TransformReplay::fullSelfReplay(new_domain, old_domain);
-
-    TORCH_INTERNAL_ASSERT(
-        old_domain->nDims() == new_domain->nDims(),
-        "Tried to set symbolic sizes through the kernel, but hit a snag, Replayed domain should be the same size as the target domain, but got ",
-        new_domain->nDims(),
-        " and ",
-        old_domain->nDims());
-    // Parallelize all iter domains
-    for (decltype(new_domain->nDims()) i{0}; i < new_domain->nDims(); i++)
-      new_domain->axis(i)->parallelize(old_domain->axis(i)->parallel_method());
-
-    tv->setDomain(new_domain);
-  }
+  fusion->setValuesMap(size_map);
+}
 
-  // Adjust memory types to make sure they are valid
-  for (TensorView* tv : all_tvs) {
-    if (fusion->hasInput(tv) || fusion->hasOutput(tv)) {
-      tv->setMemoryType(MemoryType::Global);
-    } else {
-      if (tv->getMemoryType() == MemoryType::Global)
+void IrAdjustMemoryTypes(Fusion* fusion) {
+  for (auto val : fusion->deterministic_vals()) {
+    if (ir_utils::isTV(val)) {
+      auto tv = val->as<TensorView>();
+      if (fusion->hasInput(tv) || fusion->hasOutput(tv)) {
+        tv->setMemoryType(MemoryType::Global);
+      } else if (tv->getMemoryType() == MemoryType::Global) {
         tv->setMemoryType(MemoryType::Local);
+      }
     }
   }
 }
 
 void PrepareForLowering(Fusion* fusion) {
   FusionGuard fg(fusion);
 
-  IRFixComputeAt(fusion);
-  IRValidate(fusion);
-  IRReplaceSizes();
+  IrFixComputeAt(fusion);
```

Comment:
Why?

Reply:
The main style that we seem to follow prescribes that abbreviations and acronyms to be spelled as regular words (StartRpc() rather than StartRPC()) : https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

I'm fine with deviating from this style if we document it and do it consistently (although my recommendation is to stick to the well documented Google style - if anyone has any stylistic questions we can point them there)